packages/twenty-server/src/engine/metadata-modules/agent/agent-execution.service.ts

```diff
@@ -184,31 +188,85 @@ export class AgentExecutionService {
   }
 
   private async buildUserMessageWithFiles(
-    userMessage: string,
-    fileIds?: string[],
-  ): Promise<CoreUserMessage> {
-    if (!fileIds || fileIds.length === 0) {
-      return { role: AgentChatMessageRole.USER, content: userMessage };
-    }
-
+    fileIds: string[],
+  ): Promise<(ImagePart | FilePart)[]> {
     const files = await this.fileRepository.find({
       where: {
         id: In(fileIds),
       },
     });
 
-    const textPart: TextPart = {
-      type: 'text',
-      text: userMessage,
-    };
+    return await Promise.all(files.map((file) => this.createFilePart(file)));
+  }
 
-    const fileParts = await Promise.all(
-      files.map((file) => this.createFilePart(file)),
-    );
+  private async buildUserMessage(
```

Comment:
I think this should rather be injected near AGENT_SYSTEM_PROMPTS // agent.prompt (as a system prompt), not as a user message